packages/vanilla/src/components/masthead/mastheadLeftnav.template.js

```diff
@@ -0,0 +1,133 @@
+    <nav id="${prefix}--side-nav" class="${prefix}--side-nav__navigation ${prefix}--side-nav ${prefix}--side-nav--ux ${prefix}--side-nav--hidden" aria-label="Side navigation">
+      <nav data-autoid="dds--masthead__l0-sidenav">
+        <ul class="${prefix}--side-nav__items">
+          <div class="${prefix}--side-nav__header-navigation">
+            ${_renderNav(navigation)}
+          </div>
+        </ul>
+      </nav>
+    </nav>
+  `;
+}
+
+/**
+ * Renders the array of nav link items
+ *
+ * @param {Array} links Navigation links Array
+ * @returns {string} HTML nav item
+ * @private
+ */
+function _renderNav(links) {
+  let navLinks = '';
+  if (links && links.length > 0) {
+    links.forEach(link => {
+      if (link.hasMenu) {
+        navLinks = navLinks + _renderSubnavMenu(link);
+      } else {
+        navLinks = navLinks + _renderNavItem(link);
+      }
+    });
+  }
+
+  return navLinks;
+}
+
+/**
+ * Renders a subnav nav menu
+ *
+ * @param {Array} sections Navigation links Array
+ * @returns {string} HTML nav item
+ * @private
+ */
+function _renderSubnavMenu(sections) {
+  const chevronDownIcon = toString({
+    ...chevronDown,
+    attrs: getAttributes(chevronDown.attrs),
+  });
+
+  const arrowLeftIcon = toString({
+    ...arrowLeft,
+    attrs: getAttributes(arrowLeft.attrs),
+  });
+
+  let subNavLinks = '';
+  sections.menuSections.forEach(section => {
+    section.menuItems.forEach(item => {
+      subNavLinks = subNavLinks + _renderSubNavItem(item);
+    });
+  });
+
+  return `
+    <li class="${prefix}--side-nav__item">
+      <button aria-haspopup="true" aria-expanded="false" class="${prefix}--side-nav__submenu" type="button">
+        <span class="${prefix}--side-nav__submenu-title">${sections.title}</span>
+        <div class="${prefix}--side-nav__icon ${prefix}--side-nav__icon--small ${prefix}--side-nav__submenu-chevron">
+          ${chevronDownIcon}
+        </div>
+      </button>
+      <ul class="${prefix}--side-nav__menu" role="menu">
+        <li class="${prefix}--side-nav__menu-item ${prefix}--masthead__side-nav--submemu-back" role="none">
+          <a href="javascript:void(0);" data-autoid="dds--masthead__l0-sidenav--subnav-back-1" class="${prefix}--side-nav__link" role="menuitem">
+            <span class="${prefix}--side-nav__link-text">
+              ${arrowLeftIcon}
+              Back
+            </span>
+          </a>
+        </li>
+        <li class="${prefix}--masthead__side-nav--submemu-title">${sections.title}</li>
+        ${subNavLinks}
+      </ul>
+    </li>
+  `;
+}
+
+/**
+ * Renders a single nav item
+ *
+ * @param {Array} link Navigation links Array
+ * @returns {string} HTML nav item
+ * @private
+ */
+function _renderNavItem(link) {
+  return `
+    <li class="${prefix}--side-nav__item"><a href="${link.url}" data-autoid="dds--masthead__l0-sidenav--nav-2" class="${prefix}--side-nav__link"><span class="${prefix}--side-nav__link-text">${link.title}</span></a></li>
```

Comment:
`data-autoid="dds--masthead__l0-sidenav--nav-2"` is hardcoded, this should be an iteration?

Reply:
doh! yes. I did it for the top nav but forgot this one.